codex/node.nim

```diff
@@ -233,6 +224,13 @@ proc retrieve*(
 
     # Retrieve all blocks of the dataset sequentially from the local store or network
     trace "Creating store stream for manifest", cid
+
```

Comment:
Why do we need this here, is it to prevent retrieving expired datasets?

Reply:
No, whenever we store a new dataset we need to explicitly call `trackExpiry` so that all blocks within that dataset will get maintained.